### PR TITLE
Refactor Organization to Journal

### PR DESCRIPTION
**Pull request by daisieh**
Back-end fixes to refactor the entity formerly called “Organization” to “Journal.” The external API call remains the same, i.e. `curl --request GET datadryad.org/api/v1/organizations/pgenetics/`.

Test by checking the various API calls: GET/PUT/POST new journals, GET/PUT/POST new manuscripts, editing and viewing journal concepts through the GUI.

### CODE REVIEW

File: dspace/modules/api/src/main/java/org/datadryad/rest/models/Manuscript.java [hunk 1]
```diff
@@ -189,13 +186,10 @@ public Manuscript() {} // JAXB needs this
 
     public Manuscript(DryadJournalConcept journalConcept) {
         this.journalConcept = journalConcept;
-        this.setOrganization(new Organization(journalConcept));
     }
 
     public Manuscript(LegacyManuscript legacyManuscript) {
         DryadJournalConcept journalConcept = JournalUtils.getJournalConceptByJournalID(legacyManuscript.Journal_Code);
-        this.setJournalConcept(journalConcept);
```
Comment (ryscher) [inline dspace/modules/api/src/main/java/org/datadryad/rest/models/Manuscript.java:197]: Why was this line removed? I don't see any other place in this method where this.journalConcept is set.

Comment (daisieh) [inline dspace/modules/api/src/main/java/org/datadryad/rest/models/Manuscript.java:197]: Hm. I can't remember. I can put it back; I don't know why I took it out. Good catch, probably!

**Comment by ryscher** [comment]
Looks good. Merge moved to #1185
